### PR TITLE
NIFI-12151 Fixed StandardPrivateKeyService fails due to missing Bounc…

### PR DESCRIPTION
**Pull request by turcsanyip**
…yCastleProvider

# Summary

[NIFI-12151](https://issues.apache.org/jira/browse/NIFI-12151)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-services/nifi-key-service-bundle/nifi-key-service/src/test/java/org/apache/nifi/key/service/StandardPrivateKeyServiceTest.java [hunk 1]
```diff
@@ -118,6 +134,21 @@ void testGetPrivateKeyEncryptedKey() throws Exception {
...
+                AES_128_CBC,
+                AES_192_CBC,
+                AES_256_CBC,
+                DES3_CBC,
+                PBE_SHA1_RC4_128,
+                PBE_SHA1_RC4_40,
+                PBE_SHA1_3DES,
+                PBE_SHA1_2DES,
+                PBE_SHA1_RC2_128,
+                PBE_SHA1_RC2_40
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-standard-services/nifi-key-service-bundle/nifi-key-service/src/test/java/org/apache/nifi/key/service/StandardPrivateKeyServiceTest.java:148]: It is not necessary to test all of these algorithms, particularly those algorithms where the only difference is the key size. I recommend a much smaller list:
```suggestion
                AES_256_CBC,
                DES3_CBC
```

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for tracking down this issue and providing a clean resolution @turcsanyip. The code changes look good, the only recommendation is scoping down the list of algorithms tested.

File: nifi-nar-bundles/nifi-standard-services/nifi-key-service-bundle/nifi-key-service/src/test/java/org/apache/nifi/key/service/StandardPrivateKeyServiceTest.java (hunk 1 above)
Comment (turcsanyip) [reply nifi-nar-bundles/nifi-standard-services/nifi-key-service-bundle/nifi-key-service/src/test/java/org/apache/nifi/key/service/StandardPrivateKeyServiceTest.java:148]: Thanks @exceptionfactory for your review!

Yes, it makes sense to narrow down the encryption algorithms to be tested.
However, I would keep one more: `PBE_SHA1_3DES`
This was the original test input and it ends up in the [PKCS12](https://github.com/bcgit/bc-java/blob/7349980f39aa26541f69d99e66416e84b9f7c005/pkix/src/main/java/org/bouncycastle/openssl/jcajce/JceOpenSSLPKCS8DecryptorProviderBuilder.java#L101) branch, so it is handled differently than `AES_256_CBC` and `DES3_CBC`.

**Review by exceptionfactory — APPROVED** [review]
Thanks for adjusting the set of algorithms @turcsanyip, I agree with the list included. +1 merging

**Pull request closed without merging**